src/providers/user-data/user-data.ts

```diff
@@ -240,8 +196,19 @@ export class UserDataProvider {
     this.currentWallet = undefined;
   }
 
+  public getCurrentProfile(): Profile {
+    return this.profiles[this.authProvider.loggedProfileId];
+  }
+
   loadProfiles() {
-    return this.storageProvider.getObject(constants.STORAGE_PROFILES);
+    return this.storageProvider
+      .getObject(constants.STORAGE_PROFILES)
+      .map(profiles => {
+        // we have to create "real" contacts here, because the "address" property was not on the contact object
+        // in the first versions of the app
+        Object.keys(profiles || {}).forEach(id => UserDataProvider.mapContacts(profiles[id].contacts));
```

Comment:
You can use lodash to manipulate the object, something like:

```
lodash.mapValues(profiles, profile => {
  profile.contacts = lodash.transform(profile.contacts, (acc, contact, address) => {
    contact['address'] = address;
    acc[address] = contact;
  }, {})
  
  return profile;
})
```

Reply:
Changed it! Though I personally find it much easier to read without using lodash.
But it's probably a matter of taste.